File: cpp/392 - Is Subsequence.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        // var
        int index = 0;
        
        for (int i = 0; i < s.length(); i++) {
            // var
            size_t next = t.find(s[i], index);
            
            if (next == string::npos)
                return false;
            
            index = next + 1;
        }
        return true;
    }
};

File: python/392 - Is Subsequence.py
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        
        j = 0
        for i in s:
            if i in t[j:]:
                j += t[j:].index(i) + 1
            else:
                return False
        return True
        